src/fsharp/vs/SimpleServices.fs

```diff
@@ -285,10 +287,13 @@ namespace Microsoft.FSharp.Compiler.SimpleSourceCodeServices
             let assemblyBuilderRef = ref None
             let tcImportsCapture = Some (fun tcImports -> tcImportsRef := Some tcImports)
 
-            let debugInfo =  otherFlags |> Array.exists (fun arg -> arg = "-g" || arg = "--debug:+" || arg = "/debug:+")
             // Function to generate and store the results of compilation 
+#if FX_NO_APP_DOMAINS
+            let dynamicAssemblyCreator = None
+#else
+            let debugInfo =  otherFlags |> Array.exists (fun arg -> arg = "-g" || arg = "--debug:+" || arg = "/debug:+")
```

Comment:
The ``CompileToDynamicAssembly`` methods aren't going to work if ``dynamicAssemblyCreator`` is ``None``.  Maybe elide the methods from the .NET Core API if they aren't working?